Sort Order "Comments History" tab comments correctly when same timestamp

PromInc opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->
Sort comment history on the admin order **Comments History** tab by `entity_id` if the timestamps are the exact same second.

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

### Related Pull Requests
<!-- related pull request placeholder -->

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. Fixes magento/magento2#34304

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. Create an order with two records in the `sales_order_status_history` table with the exact same timestamp.
2. View the order in the admin and select the **Comments History** tab.  The comments should be sorted correctly by `created_at` and `entity_id`

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

PromInc committed:
Correctly label option for Protection Type

The `Password Reset Protection Type` option `By IP and Email` is inaccurate as the function does and `OR` lookup, not an `AND`.

PromInc committed:
Correctly label option for Protection Type

The `Password Reset Protection Type` option `By IP and Email` is inaccurate as the function does and `OR` lookup, not an `AND`.

PromInc committed:
Correctly label option for Protection Type

The `Password Reset Protection Type` option `By IP and Email` is inaccurate as the function does and `OR` lookup, not an `AND`.

PromInc committed:
Correctly label option for Protection Type

The `Password Reset Protection Type` option `By IP and Email` is inaccurate as the function does and `OR` lookup, not an `AND`.

PromInc committed:
Correctly label option for Protection Type

The `Password Reset Protection Type` option `By IP and Email` is inaccurate as the function does and `OR` lookup, not an `AND`.

PromInc committed:
Correctly label option for Protection Type

The `Password Reset Protection Type` option `By IP and Email` is inaccurate as the function does and `OR` lookup, not an `AND`.

PromInc committed:
Correctly label option for Protection Type

The `Password Reset Protection Type` option `By IP and Email` is inaccurate as the function does and `OR` lookup, not an `AND`.

PromInc committed:
Correctly label option for Protection Type

The `Password Reset Protection Type` option `By IP and Email` is inaccurate as the function does and `OR` lookup, not an `AND`.

PromInc committed:
Sort by timestamp and id

When time stamps are equal, sort by `entity_id`.

Technically speaking this isn't perfect as well as the `entity_id`'s could come from different record sources (ie. comments and invoices).  But given that they were at the exact same time this is acceptable.  This at least ensures that the records from the same table are sorted accurately.

PromInc committed:
Merge branch 'magento:2.4-develop' into 2.4-develop

PromInc committed:
Update ResetMethod.php

PromInc committed:
Update ResetMethod.php

PromInc committed:
Update CollectionFactory.php

PromInc committed:
Update ResetMethodTest.php

PromInc committed:
Update ResetMethodTest.php

PromInc committed:
Update CollectionFactoryTest.php

PromInc committed:
Update FrequencyTest.php

PromInc committed:
Update QuantityTest.php

PromInc committed:
Update SecurityManagerTest.php

PromInc committed:
Update en_US.csv

PromInc committed:
Update ResetMethodTest.php

PromInc committed:
Force invoices to be after other records

Force invoices to go after other records.  Since the IDs don't correlate (different sources) they needed to be forced after the order was made.

ihor-sviziev commented on app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php:
```diff
@@ -303,6 +312,13 @@ public static function sortHistoryByTimestamp($a, $b)
+        if( $createdAtA->getTimestamp() === $createdAtB->getTimestamp() ) {
+            if( $a['comment'] == 'Invoice created' || $b['comment'] == 'Invoice created' ) {
+                return ( $a['comment'] == 'Invoice created' ? 1 : -1 );
+            }
```
Could you explain why do we need to have this if block?

PromInc replied:
Of course!

Here is an example screenshot of when the bug exists.  There are 3 records that occurred at the exact same time.  But they are out of order.
![image](https://user-images.githubusercontent.com/7319505/136959994-63b818c9-e03e-4523-8182-ed88c120f153.png)

We know from the codebase that Magento creates the order first and then the invoice.  So it makes sense that the order statuses are listed first and then the invoice record is last.

Because I use the Kount module, it created 2 records about the order status at the exact same time and are not sorted correctly (based on `entity_id`).  The `Processing` status should be first, then the `Review` record, and lastly the `Invoice` record.

The order records (`Processing` and `Review`) both live in table `sales_order_status_history` and thus their `entity_id`'s are sequential and able to be sorted correctly.  However the `Invoice` record doesn't live in that table and will naturally have an `entity_id` out of sequence.

Thus wehn the above line to sort by comment `Invoice created` is omitted you get this result.
![image](https://user-images.githubusercontent.com/7319505/136961008-8339cae4-89e3-494e-b052-2b783db06a52.png)

Notice that invoice if first.  The `sales_order_status_history` will typically contain more than one record per order thus those `entity_id`'s should always be greater than the `entity_id` coming from table `sales_invoice`.  So this should be a safe assumption/method to detect how to put the invoice AFTER the order information.  (not perfect, but given the mis-matched information should suffice)

Once this if block is added, the `Invoice` record is now sorted after the `Order` records as it should be and we get the proper depiction of history as it should be.
![image](https://user-images.githubusercontent.com/7319505/136961546-b01716ab-adb6-47a9-b5fe-0044987d6839.png)

ihor-sviziev replied:
Ok. It was still unclear that order history records are shown from older to newer from top to bottom.

The issue makes sense to me, but I'm afraid I have to disagree with the implementation. It doesn't make any sense to compare the entity_id of order and shipment records, but it makes sense competing for two records of the same type.

As for me, we should sort the records with the same time in the following order, similar as they're added:
- order history  https://github.com/magento/magento2/blob/6e79dc6d55a70f3a74ed4c64e6f2fa8a6ac396e6/app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php#L77
- credit memo history https://github.com/magento/magento2/blob/6e79dc6d55a70f3a74ed4c64e6f2fa8a6ac396e6/app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php#L87
- shipment history https://github.com/magento/magento2/blob/6e79dc6d55a70f3a74ed4c64e6f2fa8a6ac396e6/app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php#L106
- invoice history https://github.com/magento/magento2/blob/6e79dc6d55a70f3a74ed4c64e6f2fa8a6ac396e6/app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php#L125
- tracking numbers history https://github.com/magento/magento2/blob/6e79dc6d55a70f3a74ed4c64e6f2fa8a6ac396e6/app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php#L145

ihor-sviziev replied:
Also, the method `sortHistoryByTimestamp` isn't designed to sort records differently than by timestamps.

So I think a better solution there would be adding a new private method, `sortHistory`, that will do the following:
1. call `sortHistoryByTimestamp` to compare timestamps (for backward compatibility)
2. if the timestamp comparison result is the same for two records - try to compare by record type (see list above). 
Note: for easier comparison - I think we might introduce private constants and pass them to the "history" array like this:
```
private const HISTORY_TYPE_ORDER = 1;
private const HISTORY_TYPE_CREDIT_MEMO = 2;
private const HISTORY_TYPE_SHIPMENT = 3;
private const HISTORY_TYPE_INVOICE = 4;
private const HISTORY_TYPE_TRACKING_NUM = 5;
```
3. if the record type comparison result is the same for two records - compare by entity_id.

What do you think about it?

PromInc replied:
You're right that comparing `entity_id` for different entity types.  That is a flaw in my proposed solution - I'll admit that.  Theoretically it shouldn't be an issue, but wise to program to avoid it.

> we should sort the records with the same time in the following order, similar as they're added:
Isn't the order of the functions you outlined the same as it currently is?  And yet the issue exists?

1. You bring up a good point about the function naming and what it does.  I'm open to the final solution using a different function name.
2. I'm not fully following the need for the new constants.  I think seeing them in code might help me to better understand your intention.
3. This makes sense to me.

ihor-sviziev commented on app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php:
```diff
@@ -303,6 +312,13 @@ public static function sortHistoryByTimestamp($a, $b)
         $createdAtA = $a['created_at'];
         $createdAtB = $b['created_at'];
+        
+        if( $createdAtA->getTimestamp() === $createdAtB->getTimestamp() ) {
```
```suggestion
        if ($createdAtA->getTimestamp() === $createdAtB->getTimestamp()) {
```

PromInc committed:
Update app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php

Co-authored-by: Ihor Sviziev <ihor-sviziev@users.noreply.github.com>

ihor-sviziev commented on app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php:
```diff
@@ -76,6 +76,7 @@ public function getFullHistory()
         $history = [];
         foreach ($order->getAllStatusHistory() as $orderComment) {
             $history[] = $this->_prepareHistoryItem(
+                $orderComment->getId(),
```
It looks like it won't work w/o modifying the `_prepareHistoryItem` method.
I think even adding additional arguments to the end would be a backward incompatible change. Only introducing a new private method will be possible for doing so.

PromInc replied:
Good catch on this.  Looking back on the patch I'm running I did create a new method for `_prepareHistoryItem` since I added the ID.  I missed that when making this PR.  Look for that in an upcoming change.

ihor-sviziev commented:
Hi @PromInc,
Will you be able to update your PR with suggested changes?

PromInc committed:
Add type for sorting

Added a new function for preparing that accounts for ID and type.

The type is accounted into the sorting methods now.

A new function for sorting is added since we are now sorting by more than timestamp.

ihor-sviziev commented on app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php:
```diff
@@ -211,13 +239,30 @@ public function getItemComment(array $item)
+     * @param int $type
+     * @return array
+     */
+    protected function _prepareHistoryItemWithId($id, $label, $notified, $created, $comment = '', $type = 0)
```
Please declare argument and return types like this:
```suggestion
    private function prepareHistoryItemWithId(int $id, string $label, bool $notified, \DateTimeInterface $created, string $comment = '', int $type = 0): array
```

PromInc replied:
This is missing the leading `_` of the function name.  But these changes have been committed via commit d4c0ba5680698ac002c5c9907982d81a5be2b7a7

ihor-sviziev commented on app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php:
```diff
@@ -303,10 +348,39 @@ public static function sortHistoryByTimestamp($a, $b)
     {
         $createdAtA = $a['created_at'];
         $createdAtB = $b['created_at'];
+        
+        if ($createdAtA->getTimestamp() === $createdAtB->getTimestamp()) {
+            return $a['entity_id'] <=> $b['entity_id'];
+        }
```
Please revert this change
```suggestion
```

PromInc replied:
Good catch...  my bad.

ihor-sviziev commented on app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php:
```diff
@@ -303,10 +348,39 @@ public static function sortHistoryByTimestamp($a, $b)
+     * @param mixed $b
+     * @return int
+     */    
+    private function sortHistory($a, $b)
```
```suggestion
    private function sortHistory(array $a, array $b): int
```

ihor-sviziev commented on app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php:
```diff
@@ -303,10 +348,39 @@ public static function sortHistoryByTimestamp($a, $b)
+     * Comparison For Sorting History By Timestamp and entity_id
+     *
+     * @param mixed $a
+     * @param mixed $b
```
```suggestion
     * @param array $a
     * @param array $b
```

ihor-sviziev commented on app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php:
```diff
@@ -303,10 +348,39 @@ public static function sortHistoryByTimestamp($a, $b)
+     */    
+    private function sortHistory($a, $b)
+    {
+        $createdAtA = $a['created_at'];
+        $createdAtB = $b['created_at'];
+        
+        if ($createdAtA->getTimestamp() === $createdAtB->getTimestamp()) {
+            if ($a['type'] < $b['type']) {
+                return 1;
+            } elseif ($a['type'] > $b['type']) {
+                return -1;
+            } else {
+                return $a['entity_id'] <=> $b['entity_id'];
+            }
+        }
+
+        return $createdAtA->getTimestamp() <=> $createdAtB->getTimestamp();
```
I would suggest something like this:
```suggestion
        $result = $this->sortHistoryByTimestamp($a, $b);
        if (0 !== $result) {
            return $result;
        }
        
        $result = $a['type'] <=> $b['type'];
        if (0 !== $result) {
            return $result;
        }
        
        return $a['entity_id'] <=> $b['entity_id'];
```

ihor-sviziev commented:
@magento run Static Tests

PromInc committed:
Update app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php

Co-authored-by: Ihor Sviziev <ihor-sviziev@users.noreply.github.com>

PromInc committed:
Update app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php

Co-authored-by: Ihor Sviziev <ihor-sviziev@users.noreply.github.com>

PromInc committed:
Update app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php

Co-authored-by: Ihor Sviziev <ihor-sviziev@users.noreply.github.com>

PromInc committed:
define arguments

PromInc committed:
Update app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php

Co-authored-by: Ihor Sviziev <ihor-sviziev@users.noreply.github.com>

ihor-sviziev commented:
@magento run all tests

ihor-sviziev reviewed (changes requested):
Please fix failing tests and cover your changes with some type of automated tests

PromInc commented:
@ihor-sviziev i have no idea what is expected of me at this time.

engcom-Lima commented:
@magento run Functional Tests B2B,Functional Tests CE,Functional Tests EE,Static Tests

engcom-Lima commented:
Hi @PromInc,
Thanks for your contribution!
I am looking into the build failures.

Thanks

Commit:
Merge branch '2.4-develop' of https://github.com/magento/magento2 into 2.4-develop

Commit:
Merge branch '2.4-develop' of https://github.com/magento/magento2 into 2.4-develop

Commit:
Merge branch '2.4-develop' of https://github.com/magento/magento2 into 2.4-develop

engcom-Lima committed:
Merge branch '2.4-develop' of https://github.com/magento/magento2 into 2.4-develop

engcom-Lima commented:
@magento run all tests

engcom-Lima commented:
@magento run Static Tests, WebAPI Tests

engcom-Lima committed:
Merge branch '2.4-develop' of https://github.com/PromInc/magento2 into 2.4-develop

engcom-Lima committed:
Merge branch '2.4-develop' of https://github.com/magento/magento2 into 2.4-develop

engcom-Lima committed:
[BUGFIX] - Fixed static test failures

engcom-Lima commented:
@magento run all tests

PromInc committed:
Merge branch 'magento:2.4-develop' into 2.4-develop

PromInc committed:
Use configuration for mage-messages cookie

The mage-messages doesn't use the configuration values for the `mage-messages` cookie.

PromInc committed:
Create Config.php

PromInc committed:
Pass the configured the cookie domain to Javascript

PromInc committed:
set the cookieDomain via the config value

PromInc committed:
Delete Config.php

ihor-sviziev commented on app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php:
```diff
@@ -218,6 +246,35 @@ protected function _prepareHistoryItem($label, $notified, $created, $comment = '
+     * @param int $type
+     * @return array
+     */
+    private function _prepareHistoryItemWithId(
```
Please remove the underscore from the method name, it was an old practice from Magento 1, the new code shouldn't contain it
```suggestion
    private function prepareHistoryItemWithId(
```

PromInc replied:
@ihor-sviziev I have updated this function name and addressed the other un-related files that creeped into this pull request.  Shame on me for not following git best-practices...  Sorry this got a bit messy.  I think this is all straightened out now.

ihor-sviziev commented on app/code/Magento/Theme/Controller/Result/MessagePlugin.php:
```diff
@@ -59,6 +60,11 @@ class MessagePlugin
+    /**
+     * @var ScopeConfigInterface
+     */
+    protected $scopeConfig;
```
Please make it private
```suggestion
    private $scopeConfig;
```

ihor-sviziev commented on app/code/Magento/Theme/Controller/Result/MessagePlugin.php:
```diff
@@ -171,11 +180,36 @@ private function setCookie(array $messages)
-            $publicCookieMetadata->setDurationOneYear();
-            $publicCookieMetadata->setPath($this->sessionConfig->getCookiePath());
+
+            if( $configLifetime = $this->scopeConfig->getValue(
```
I don't think this change is related to the issue you're trying to fix. Could you please revert it?

ihor-sviziev commented on app/code/Magento/Theme/Controller/Result/MessagePlugin.php:
```diff
@@ -171,11 +180,36 @@ private function setCookie(array $messages)
             $publicCookieMetadata = $this->cookieMetadataFactory->createPublicCookieMetadata();
-            $publicCookieMetadata->setDurationOneYear();
-            $publicCookieMetadata->setPath($this->sessionConfig->getCookiePath());
+
+            if( $configLifetime = $this->scopeConfig->getValue(
+                'web/cookie/cookie_lifetime',
+                \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+            ) ) {
+                $publicCookieMetadata->setDuration($configLifetime);
+            } else {
+                $publicCookieMetadata->setDurationOneYear();
+            }
+
+            if( $configPath = $this->scopeConfig->getValue(
+                'web/cookie/cookie_path',
+                \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+            ) ) {
+                $publicCookieMetadata->setPath($configPath);
+            } else {
+                 $publicCookieMetadata->setPath($this->sessionConfig->getCookiePath());
+            }
+            
```
I don't think this change is related to the issue you're trying to fix. Could you please revert it?
```suggestion
```

PS: you can create a new branch and add them to it, so they won't affect this PR.

ihor-sviziev commented on app/code/Magento/Theme/Controller/Result/MessagePlugin.php:
```diff
@@ -171,11 +180,36 @@ private function setCookie(array $messages)
+            }
+            
             $publicCookieMetadata->setHttpOnly(false);
+
```
and this
```suggestion
```

ihor-sviziev commented on app/code/Magento/Theme/Controller/Result/MessagePlugin.php:
```diff
@@ -171,11 +180,36 @@ private function setCookie(array $messages)
+
             $publicCookieMetadata->setSameSite('Strict');
 
+            if( $configDomain = $this->scopeConfig->getValue(
+                'web/cookie/cookie_domain',
+                \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+            ) ) {
+                $publicCookieMetadata->setDomain($configDomain);
+            }
+
```
And this
```suggestion
```

ihor-sviziev commented on app/code/Magento/Theme/view/frontend/templates/messages.phtml:
```diff
@@ -4,6 +4,7 @@
  * See COPYING.txt for license details.
  */
 ?>
+<?php $helperConfig = $this->helper('Magento\Theme\Helper\Data'); ?>
```
And this
```suggestion
```

ihor-sviziev commented on app/code/Magento/Theme/view/frontend/templates/messages.phtml:
```diff
@@ -35,7 +36,8 @@
                 "components": {
                         "messages": {
-                            "component": "Magento_Theme/js/view/messages"
+                            "component": "Magento_Theme/js/view/messages",
+                            "cookieDomain": "<?= $helperConfig->getConfig( 'web/cookie/cookie_domain' ) ?>"
```
And this
```suggestion
                            "component": "Magento_Theme/js/view/messages"
```

ihor-sviziev commented on app/code/Magento/Theme/view/frontend/web/js/view/messages.js:
```diff
@@ -41,7 +41,7 @@ define([
             $.mage.cookies.set('mage-messages', '', {
                 samesite: 'strict',
-                domain: ''
+                domain: this.cookieDomain
```
And this
```suggestion
                domain: ''
```

PromInc committed:
Revert "set the cookieDomain via the config value"

This reverts commit 496302db2505797484ec4784992bf668703441d5.

PromInc committed:
Revert "Pass the configured the cookie domain to Javascript"

This reverts commit 0a9223d81103cb80d4afd9f970e6be53d3b57a05.

PromInc committed:
Revert "Use configuration for mage-messages cookie"

This reverts commit f7a746faf505c286fa65a579271838ca23cf287c.

PromInc committed:
Update function name

Remove leading _ from new function per mention by @ihor-sviziev of abandoned M1 practice

ihor-sviziev commented:
@magento run all tests

engcom-Charlie commented:
Moving back to Changes Requested as PR got moved in Review by bot after assignment.

engcom-Charlie committed:
Merge remote-tracking branch 'magento2/2.4-develop' into 2.4-develop

engcom-Charlie commented:
@magento run all tests

engcom-Charlie committed:
Issue-34304 Fixed multiple tests failurs.

engcom-Charlie commented:
Hi @PromInc , @ihor-sviziev 

Thank you for your contribution!

While going through the build failures found that there were 2 issues in given solution and because of which many test cases were failing. I have fixed both of them in latest commit.

1. /sales/order/view/order_id/{id number}/  page was broken if respective email notification fields of invoice, shipment, credit memo was set to NULL. It was showing error  and because of this many (more than 200) functional test cases were failed.

Before latest commit:
<img width="619" alt="image" src="https://user-images.githubusercontent.com/51680850/156154677-54a6546c-3ddb-49d9-a809-c073fb538a3d.png">

<img width="1433" alt="image" src="https://user-images.githubusercontent.com/51680850/156156561-57f13e5c-42e9-4edc-8878-bb171dea08f7.png">

After latest commit: 
<img width="1370" alt="image" src="https://user-images.githubusercontent.com/51680850/156157116-7f0c4135-f0da-4a21-b572-cb3b24455bae.png">

2. After fixing above issue found that because of this PR implementation some unwanted records like that Shipment created, Invoice created and Credit memo created are also getting displayed under "Notes for this Order" section from Comments History tab and which is not the current behaviour of Magento 2.4-develop.
So fixed this issue. Now "Notes for this Order" section from Comments History tab is displaying only those records which are getting displayed on 2.4-develop

Before latest commit:

![image](https://user-images.githubusercontent.com/51680850/156155998-969a1894-9b4b-44ba-8f6c-fc8d851c967d.png)


After latest commit: 

<img width="1240" alt="image" src="https://user-images.githubusercontent.com/51680850/156156088-82c2b761-6848-431b-bee0-1af266d56dd1.png">

Thank you!

engcom-Charlie commented:
Hi @ihor-sviziev,

As mentioned in [comment](https://github.com/magento/magento2/pull/34306#issuecomment-1055304393), I have modified the given solution in order to fix build failure.

Requesting for review. Can you please do the review again.

Thank you!

engcom-Charlie commented:
@magento run all tests

engcom-Charlie committed:
Merge remote-tracking branch 'magento2/2.4-develop' into 2.4-develop

engcom-Charlie commented:
@magento run all tests

engcom-Charlie commented:
@magento run all tests

engcom-Charlie commented:
With respect to previous build failure, all the failed tests mentioned in [comment](https://github.com/magento/magento2/pull/34306#issuecomment-1055304393)  are working fine now. There are some static test failures right now. Looking into that.

engcom-Charlie committed:
Issue-34304 Fixed tests failurs.

engcom-Charlie commented:
@magento run all tests

engcom-Charlie commented:
@magento run all tests

engcom-Charlie commented:
@magento run Unit Tests, Functional Tests B2B , Functional Tests CE, Functional Tests EE

engcom-Charlie commented:
@magento run Unit Tests, Functional Tests B2B , Functional Tests CE, Functional Tests EE

engcom-Charlie committed:
Issue-34304 Fixed tests failurs.

engcom-Charlie commented:
@magento run all tests

engcom-Charlie commented:
Fixed failing static as well as some functional test cases. 

Now the test case which is failing(AdminCheckVATNumberValidationButtonForEUCountryTest) in build is not part of this PR or not failing because of any changes made in this PR. As mentioned in [comment](https://github.com/magento/magento2/pull/34306#issuecomment-1055307055) requesting for the review, hence moving it to review status.

Thank you!
